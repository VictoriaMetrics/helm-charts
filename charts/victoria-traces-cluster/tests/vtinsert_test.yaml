suite: vtinsert templates test
chart:
  version: 0.1.1
  appVersion: 0.1.0
templates:
  - vtinsert-server.yaml
tests:
  - it: deployment should match snapshot
    asserts:
      - matchSnapshot: {}
  - it: deployment should match snapshot with fullnameOverride, extraLabels and podLabels
    set:
      vtinsert:
        name: insert
        fullnameOverride: vtinsert-node
        extraLabels:
          extraLabelName: extraLabelValue
        podLabels:
          podLabelName: podLabelValue
        resources:
          limits:
            cpu: 100m
            memory: 200Mi
          requests:
            cpu: 100m
            memory: 200Mi
      vtstorage:
        extraArgs:
          httpListenAddr: :10492
    asserts:
      - matchSnapshot: {}
