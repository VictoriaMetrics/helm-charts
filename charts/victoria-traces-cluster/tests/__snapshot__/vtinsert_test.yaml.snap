deployment should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: vtinsert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vt-cluster
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: victoria-traces-cluster-0.1.1
      name: RELEASE-NAME-vt-cluster-vtinsert
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          app: vtinsert
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: vt-cluster
      template:
        metadata:
          labels:
            app: vtinsert
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: vt-cluster
        spec:
          containers:
            - args:
                - --envflag.enable
                - --http.shutdownDelay=15s
                - --httpListenAddr=:10481
                - --loggerFormat=json
                - --storageNode=RELEASE-NAME-vt-cluster-vtstorage-0.RELEASE-NAME-vt-cluster-vtstorage.NAMESPACE.svc.cluster.local.:10491
                - --storageNode=RELEASE-NAME-vt-cluster-vtstorage-1.RELEASE-NAME-vt-cluster-vtstorage.NAMESPACE.svc.cluster.local.:10491
              image: victoriametrics/victoria-traces:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 3
                initialDelaySeconds: 5
                periodSeconds: 15
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vtinsert
              ports:
                - containerPort: 10481
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 5
                timeoutSeconds: 5
              securityContext:
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
          securityContext:
            fsGroup: 1000
          terminationGracePeriodSeconds: 30
deployment should match snapshot with fullnameOverride, extraLabels and podLabels:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app: insert
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: vt-cluster
        app.kubernetes.io/version: 0.1.0
        extraLabelName: extraLabelValue
        helm.sh/chart: victoria-traces-cluster-0.1.1
      name: vtinsert-node
      namespace: NAMESPACE
    spec:
      replicas: 2
      selector:
        matchLabels:
          app: insert
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: vt-cluster
      template:
        metadata:
          labels:
            app: insert
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: vt-cluster
            podLabelName: podLabelValue
        spec:
          containers:
            - args:
                - --envflag.enable
                - --http.shutdownDelay=15s
                - --httpListenAddr=:10481
                - --loggerFormat=json
                - --storageNode=RELEASE-NAME-vt-cluster-vtstorage-0.RELEASE-NAME-vt-cluster-vtstorage.NAMESPACE.svc.cluster.local.:10492
                - --storageNode=RELEASE-NAME-vt-cluster-vtstorage-1.RELEASE-NAME-vt-cluster-vtstorage.NAMESPACE.svc.cluster.local.:10492
              image: victoriametrics/victoria-traces:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                failureThreshold: 3
                initialDelaySeconds: 5
                periodSeconds: 15
                tcpSocket:
                  port: http
                timeoutSeconds: 5
              name: vtinsert
              ports:
                - containerPort: 10481
                  name: http
              readinessProbe:
                failureThreshold: 3
                httpGet:
                  path: /health
                  port: http
                  scheme: HTTP
                initialDelaySeconds: 5
                periodSeconds: 5
                timeoutSeconds: 5
              resources:
                limits:
                  cpu: 100m
                  memory: 200Mi
                requests:
                  cpu: 100m
                  memory: 200Mi
              securityContext:
                runAsGroup: 1000
                runAsNonRoot: true
                runAsUser: 1000
          securityContext:
            fsGroup: 1000
          terminationGracePeriodSeconds: 30
